### DISCUSSION TITLE
Automated schema validation scripts

### DISCUSSION THREAD

**Discussion by ozarkblue** [Apicurio/apicurio-registry#2951] [Ideas]
Anyone already using/familier with below please let me know - will be helpful.
I'm trying to perform some schema validation before uploading schema in registry.
Other than syntax check (which his already part of Apicurio) like to perform some more custom  rules to ensure overall schema structure/content health is consistant and compatible with schema governance perspective.
Some of such rules as below:

- All fields should be having description
- certain naming standards are followed in fields
- Tags are added 
- if any property added check that with some predefined constraints say team contact email should follow email format.Pr product team ID should match a format etc

While coding with any of the programing language is always an option but  like to have more of a regular expression based scripting tool that way configuration of these validation rules will be simpler.

I'm in process of google search now to fins a right one. If anyone knows any script based rule scrips (preferably open source and light weight)  - Pls let me know. Appreciate in advance.

@EricWittmann  any suggestion ?

**Comment by EricWittmann** [comment]
Hey @ozarkblue sorry for the late response.  We are looking into this same sort of functionality in both API Designer and in Registry.  It is **probable** that we will use Spectral for this.

https://stoplight.io/open-source/spectral
